networkdb/networkdb_test.go

```diff
@@ -826,7 +829,6 @@ func TestNetworkDBIslands(t *testing.T) {
 	for i := 0; i < 3; i++ {
 		logrus.Infof("node %d leaving", i)
 		dbs[i].Close()
-		time.Sleep(2 * time.Second)
```

Comment:
perhaps in addition to this change ( to me, it is good to remove unneeded sleep), we should do a polling loop around line 836-839?

Reply:
A poll that checks, every x seconds and breaks after 10 seconds, if these conditions are satisfied
```
dbs[i].leftNodes ==  && dbs[i].failedNodes == 0
```
  not a showstopper but best case scenario it will speed up the test.

Note, the sleep approach is used throughout the code, would be nice if we can refactor to start polling instead

Reply:
ack